pkg/dogstatsd/server.go

```diff
@@ -196,80 +197,86 @@ func (s *Server) worker(metricOut chan<- *metrics.MetricSample, eventOut chan<-
 		case <-s.stopChan:
 			return
 		case <-s.health.C:
-		case packet := <-s.packetIn:
-			extraTags := s.extraTags
-
-			if packet.Origin != listeners.NoOrigin {
-				log.Tracef("Dogstatsd receive from %s: %s", packet.Origin, packet.Contents)
-				originTags, err := tagger.Tag(packet.Origin, tagger.DogstatsdCardinality)
-				if err != nil {
-					log.Errorf(err.Error())
-				} else {
-					extraTags = append(extraTags, originTags...)
-				}
-				log.Tracef("Tags for %s: %s", packet.Origin, originTags)
-			} else {
-				log.Tracef("Dogstatsd receive: %s", packet.Contents)
+		case packets := <-s.packetsIn:
+			for _, packet := range packets {
+				s.handlePacket(packet, metricOut, eventOut, serviceCheckOut)
 			}
+		}
+	}
+}
 
-			for {
-				message := nextMessage(&packet.Contents)
-				if message == nil {
-					break
-				}
+func (s *Server) handlePacket(packet *listeners.Packet, metricOut chan<- *metrics.MetricSample, eventOut chan<- metrics.Event, serviceCheckOut chan<- metrics.ServiceCheck) {
+	extraTags := s.extraTags
```

Comment:
Sanity check: tags added to `extraTags` in this method won't be added to `s.extraTags`, seeing the code I guess this is the intended behavior? Just to be sure.

Reply:
Nothing changed in this method anyway but yes this seems correct. We add the origin which is specific to a packet